sql/core/src/test/resources/sql-tests/analyzer-results/time.sql.out

```diff
@@ -521,3 +521,92 @@ org.apache.spark.sql.catalyst.ExtendedAnalysisException
     "fragment" : "'10:00:01' :: TIME(1) - INTERVAL '1' MONTH"
   } ]
 }
+
+
+-- !query
+SELECT TIME'12:30:41' - TIME'10:00'
+-- !query analysis
+Project [12:30:41 - 10:00:00 AS TIME '12:30:41' - TIME '10:00:00'#x]
++- OneRowRelation
+
+
+-- !query
+SELECT TIME'08:30' - NULL
```

Comment:
Why is this not NULL?
Vast majority of builtin's are "return NULL on NULL input". I'm sure this is true for arithmetic operators in general.

Reply:
I think it's consistent with `TIMESTAMP'xxx' - NULL`, but we can always match the NullType explicitly to support this case. Maybe we can discuss it in a followup PR?

Reply:
Here is the PR: https://github.com/apache/spark/pull/51454 which changes the behaviour.